tensorflow/contrib/data/python/ops/resampling.py

```diff
@@ -50,70 +51,171 @@ def rejection_resample(class_func, target_dist, initial_dist=None, seed=None):
     A `Dataset` transformation function, which can be passed to
     @{tf.data.Dataset.apply}.
   """
-
   def _apply_fn(dataset):
     """Function from `Dataset` to `Dataset` that applies the transformation."""
-    dist_estimation_batch_size = 32
     target_dist_t = ops.convert_to_tensor(target_dist, name="target_dist")
     class_values_ds = dataset.map(class_func)
+
+    # Get initial distribution.
     if initial_dist is not None:
-      initial_dist_t = ops.convert_to_tensor(initial_dist, name="initial_dist")
+      initial_dist_t = ops.convert_to_tensor(
+          initial_dist, name="initial_dist")
       acceptance_dist = _calculate_acceptance_probs(initial_dist_t,
                                                     target_dist_t)
       initial_dist_ds = dataset_ops.Dataset.from_tensors(
           initial_dist_t).repeat()
       acceptance_dist_ds = dataset_ops.Dataset.from_tensors(
           acceptance_dist).repeat()
     else:
-      num_classes = (target_dist_t.shape[0].value or
-                     array_ops.shape(target_dist_t)[0])
-      smoothing_constant = 10
-      initial_examples_per_class_seen = array_ops.fill(
-          [num_classes], np.int64(smoothing_constant))
-
-      def update_estimate_and_tile(num_examples_per_class_seen, c):
-        updated_examples_per_class_seen, dist = _estimate_data_distribution(
-            c, num_examples_per_class_seen)
-        tiled_dist = array_ops.tile(
-            array_ops.expand_dims(dist, 0), [dist_estimation_batch_size, 1])
-        return updated_examples_per_class_seen, tiled_dist
-
-      initial_dist_ds = (class_values_ds.batch(dist_estimation_batch_size)
-                         .apply(scan_ops.scan(initial_examples_per_class_seen,
-                                              update_estimate_and_tile))
-                         .apply(batching.unbatch()))
+      initial_dist_ds = _estimate_initial_dist_ds(
+          target_dist_t, class_values_ds)
       acceptance_dist_ds = initial_dist_ds.map(
           lambda initial: _calculate_acceptance_probs(initial, target_dist_t))
+    return _filter_ds(dataset, acceptance_dist_ds, initial_dist_ds,
+                      class_values_ds, seed)
+
+  return _apply_fn
+
+
+def rejection_resample_v2(class_func, target_dist, initial_dist=None,
+                          seed=None):
```

Comment:
Should we just replace the old version with this, and call it `rejection_resample()`?